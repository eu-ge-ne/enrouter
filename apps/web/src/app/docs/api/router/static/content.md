# StaticRouter API

## Components

### StaticRouter

```typescript
interface StaticRouterProps {
  handlers: RouteHandler;
  location: string;
  matches: RouteMatch[];
}

function StaticRouter({ handlers, location, matches }: StaticRouterProps);
```
