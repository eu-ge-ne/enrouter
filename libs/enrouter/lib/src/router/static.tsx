import type { ReactNode } from "react";

import type { Match } from "#lib/match/mod.js";
import { NavigateProvider, noNavigate } from "#lib/navigate/mod.js";
import { LocationProvider } from "#lib/location/mod.js";
import { MatchProvider } from "#lib/match/context.js";
import { Root } from "./root.js";

export interface StaticProps {
  location: string;
  match: Match | undefined;
}

export function Static({ location, match }: StaticProps): ReactNode {
  return (
    <NavigateProvider value={noNavigate}>
      <LocationProvider value={location}>
        <MatchProvider value={match}>
          <Root />
        </MatchProvider>
      </LocationProvider>
    </NavigateProvider>
  );
}
