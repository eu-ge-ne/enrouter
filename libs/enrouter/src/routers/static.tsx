import { useMemo } from "react";

import { renderMatches } from "#render/mod.js";
import { RouterContext } from "./context.js";

import type { RouteHandler } from "#handlers/mod.js";
import type { RouteMatch } from "#matches/mod.js";
import type { TRouterContext } from "./context.js";

interface StaticRouterParams {
  handlers: RouteHandler;
  location: string;
  matches: RouteMatch[];
}

export function StaticRouter({
  handlers,
  location,
  matches,
}: StaticRouterParams) {
  const context = useMemo<TRouterContext>(
    () => ({
      handlers,
      location,
      navigate: () => undefined,
    }),
    [handlers, location],
  );

  const children = useMemo(() => renderMatches(matches), [matches]);

  return (
    <RouterContext.Provider value={context}>{children}</RouterContext.Provider>
  );
}
