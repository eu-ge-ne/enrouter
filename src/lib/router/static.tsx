import type { ReactNode, ReactElement } from "react";

import type { Match } from "#lib/match/match.js";
import { noNavigate } from "#lib/navigate/mod.js";
import { Root } from "./root.js";

export interface StaticRouterProps {
  root?: ReactElement;
  location: string;
  matches: Match[];
}

export function StaticRouter(props: StaticRouterProps): ReactNode {
  return <Root {...props} navigate={noNavigate} />;
}
