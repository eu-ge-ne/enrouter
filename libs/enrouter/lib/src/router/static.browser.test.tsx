import type { FC, PropsWithChildren } from "react";
import { describe, test, expect } from "vitest";
import { render } from "vitest-browser-react";
import * as regexparam from "regexparam";

import { StaticRouter } from "./static.js";
import type { Route } from "#lib/route/mod.js";
import type { Match } from "#lib/match/mod.js";

const wrapperId = "test-wrapper";

const wrapper: FC<PropsWithChildren> = ({ children }) => (
  <div data-testid={wrapperId}>{children}</div>
);

describe("router/static", () => {
  test("1 route and 1 match", async () => {
    const routes: Route = {
      path: "/",
      test: regexparam.parse("/", true),
      modules: [],
      loaded: true,
      elements: {
        layout: {
          main: <div>Layout</div>,
        },
      },
    };

    const matches: Match[] = [
      {
        route: routes,
        location: "/",
        isFull: true,
        params: {},
      },
    ];

    const screen = render(
      <StaticRouter routes={routes} location="/" matches={matches} />,
      {
        wrapper,
      },
    );

    await expect.element(screen.getByTestId(wrapperId)).toBeVisible();
    expect(screen.container).toMatchSnapshot();
  });
});
