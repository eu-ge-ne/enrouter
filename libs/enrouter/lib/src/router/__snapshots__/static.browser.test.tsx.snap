// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`router/static > 1 route and 1 match 1`] = `
<div>
  <div
    data-testid="test-wrapper"
  >
    <div>
      Layout
    </div>
  </div>
</div>
`;
